v4/src/EditNodeModal.jsx

```diff
@@ -0,0 +1,240 @@
+// eslint-disable-line
+/* eslint-disable no-alert */
+/* eslint-disable no-nested-ternary */ // XXX: !!!!!!!!PLEASE FIX THIS!!!!!!!
+import React, { Component } from 'react';
+import {
+  Button, Modal, Form, ToggleButtonGroup, ToggleButton,
+} from 'react-bootstrap';
+
+import { PropTypes } from 'prop-types';
+import { NTYPE, RTYPE } from './constants';
+
+import './styles/AddNodeModal.css';
```

Comment:
This might get a bit messy, with overlapping dependencies. You should write a separate `EditNodeModal.css` file or inherit some style from parent components